### DISCUSSION TITLE
Paths from configuration file have been overridden by paths provided as command arguments.

### DISCUSSION THREAD

**Discussion by Leksizz** [PHP-CS-Fixer/PHP-CS-Fixer#7068] [Q&A] [answered]
Hello everyone I am new to programming and decided to install php-cs-fixer. When using the command php-cs-fixer fix an error is displayed: Paths from configuration file have been overridden by paths provided as command arguments. Fixed 0 of 2 files in 0.002 seconds, 12.000 MB memory used

Files that were not fixed due to errors reported during linting before fixing: [File Path index.php].
Please tell me what could be wrong, it may be necessary to attach the file configuration .php-cs-fixer.php or setting.json?

**Comment by Wirone** [comment]
Hi @Leksizz and welcome to the Fixer users group! This warning is shown when you provide paths to files/directories in the CLI command when running Fixer and this is expected behavior.

**Reply by Leksizz** [reply]
And how do I correct my files using this extension?

**Reply by Wirone** [reply]
Everything is [in the docs](https://cs.symfony.com/doc/config.html). Basically you rather want to define covered paths in config file and override them through CLI arguments only when you want to narrow analysis. But you can also take advantage of cache and run always for all configured files.

**Comment by SpacePossum** [comment]
Hi!
Thanks for trying out to tool!

The error you get indicates that you provide information about which files you want to fix in two ways. Because you provided two ways; one time through the command you are using and one time through a configuration file, the tool does not understand what you want to do. 

If you can provide the command that you use and the configuration file you use and what you want to happen we might be able to assist you further.

**Reply by Leksizz** [reply]
Thanks for the answer! I would like to fix my file index.php
 php-cs-fixer fix C:\OSPanel\domains\test
Loaded config default from "C:\OSPanel\domains\test\.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
Paths from configuration file have been overridden by paths provided as command arguments.

Fixed 0 of 2 files in 0.002 seconds, 12.000 MB memory used

Files that were not fixed due to errors reported during linting before fixing:
   1) C:\OSPanel\domains\test\index.php
   I attach screenshots of the files
.php-cs-fixer.php
![screen  php-cs-fixer php 1](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/129582133/15137805-83e6-4899-b5c0-a9acbb3b536c)
![screen  php-cs-fixer php 2](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/129582133/cb58d76a-f818-4cc4-a329-7a8064729fc3)
index.php
![screen index php](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/129582133/5ce60049-b757-492d-bf8c-08c585efb977)
settings.json
![screen settings json](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/129582133/0c2feb30-5614-4691-97b7-ac00f92dfdbd)

**Reply by SpacePossum** [reply, marked as answer]
Nice, than you have two options:
- to fix just `C:\OSPanel\domains\test\index.php` you can use `php-cs-fixer fix C:\OSPanel\domains\test\index.php -vvv`
- or, if you want to fix any `.php` file in the directory `C:\OSPanel\domains\test` you can use `php-cs-fixer fix -vvv`

There is however a syntax error in your `index.php` so the tool will not touch it. Might be best to fix that file before trying to run the tool again. It is likely because of `$req-&gt;getHeader` which I assume should be `$req->getHeader` , maybe it comes from a copy-paste of some HTML.

**Reply by Leksizz** [reply]
When entering outputs:
php-cs-fixer fix C:\OSPanel\domains\test\index.php -vvv
PHP CS Fixer 3.18.0 Triangle by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.1.9
Loaded config default from "C:\OSPanel\domains\test\.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
Paths from configuration file have been overridden by paths provided as command arguments.
I                                                                                                           1 / 1 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error

Fixed 0 of 1 files in 0.001 seconds, 12.000 MB memory used

Files that were not fixed due to errors reported during linting before fixing:
   1) C:\OSPanel\domains\test\index.php


        [PhpCsFixer\Linter\LintingException]
        Parse error: syntax error, unexpected token ";", expecting identifier or variable or "{" or "$" on line 17.


Detected deprecations in use:
- Option "tokens: use_trait" used in `no_extra_blank_lines` rule is deprecated, use the rule `class_attributes_separation` with `elements: trait_import` instead.
- Rule "braces" is deprecated. Use "single_space_around_construct", "control_structure_braces", "curly_braces_position", "control_structure_continuation_position", "declare_parentheses", "statement_indentation", "no_multiple_statements_per_line" and "no_extra_blank_lines" instead.

**Reply by Wirone** [reply]
There's `$req-&gt;` instead of `$req->`, because most probably you copy-pasted it from somewhere. It's not a problem with a tool, but typical PEBKAC 😉. We really would like to help you with Fixer usage, but we can't guide you through basics of programming. IDE underlines problems and reports them, just read the feedback and fix the problems within your code.

**Reply by Leksizz** [reply]
Thanks for the answer! The code in the file is copied to check how the fixer can fix it. But for some reason it is not possible to do this.

**Reply by Wirone** [reply]
You've been said why it's impossible 2 times already ([1](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7068#discussioncomment-6220638), [2](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7068#discussioncomment-6220715)), but let's point it again: you have **syntax error**, instead of `$req-&gt;` there should be `$req->` (in 2 places), also `try` block is not closed and there's some `.....` on line 27. Fixer is unable to properly tokenize your code, so analysis can't be done. Fix the code (no red curly underlines in IDE 😉), and re-run the Fixer.

**Reply by Leksizz** [reply]
If I understand you correctly, the fixer cannot edit my code, since some of its points first need to be edited manually?

**Reply by Wirone** [reply]
Yes, Fixer can analyse and fix only valid PHP code.

**Reply by SpacePossum** [reply]
Yes, the fixer can only fix code that has no syntax issues.
You can try for yourself what PHP thinks of the code using this command:
>php -l C:\OSPanel\domains\test\index.php

**Reply by Leksizz** [reply]
@Wirone @SpacePossum Thank you very much for your help and patience!

**Reply by SpacePossum** [reply]
you're welcome, good luck and have fun programming :D
